app/controllers/resources_controller.rb

```diff
@@ -59,7 +59,25 @@ def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    result = assumed_role.allowed_to?(privilege, resource)
+    begin
+      result = assumed_role.allowed_to?(privilege, resource)
+    rescue
+      result = false
+      resource_id = "#{params[:account]}:#{params[:kind]}:#{params[:identifier]}"
+      role_id = params[:role]
+
```

Comment:
Perhaps once setting those parameters above, the already existing `Audit.logger.log` can use them, such that there is no need to call it again?

Reply:
`Audit.logger.log` needs to be called twice anyway, because when there is an exception (like in the failure scenarios) there exception should be raised again after writing the audit. The second `Audit.logger.log` is for scenarios of no exceptions when the program should keep running